error.go

```diff
@@ -9,6 +9,12 @@ import (
 // https://kafka.apache.org/protocol#protocol_error_codes
 type Error int
 
+// ErrorWithCode retains Error code information while creating an error message
+type ErrorWithCode struct {
+	Error error
+	Code  int16
+}
```

Comment:
See my response to this discussion https://github.com/segmentio/kafka-go/pull/596#discussion_r576451053

I don't think we need to introduce this new public API.

Reply:
Done.